subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractSourceIncrementalCompilationIntegrationTest.groovy

```diff
@@ -1079,8 +1079,15 @@ sourceSets {
 
     def "recompiles all classes in a package if the package-info file changes"() {
         given:
+        file("src/main/${languageName}/annotations/Anno.${languageName}").text = """
+            package annotations;
+            import java.lang.annotation.*;
+            @Retention(RetentionPolicy.RUNTIME)
+            @Target(ElementType.PACKAGE)
+            public @interface Anno {}
+        """
         def packageFile = file("src/main/${languageName}/foo/package-info.${languageName}")
-        packageFile.text = """package foo;"""
+        packageFile.text = """@Deprecated package foo;"""
```

Comment:
Yes, but then the test is the same as "adding a package-info", because a package-info file without any annotations is not compiled into a class file at all.